primefaces/src/main/java/org/primefaces/model/filter/ComparableFilterConstraint.java

```diff
@@ -38,35 +34,18 @@ public boolean isMatching(FacesContext ctxt, Object value, Object filter, Locale
         if (value == null || filter == null) {
             return false;
         }
-        if (value instanceof Number) {
-            BigDecimal filterBigDecimal = toBigDecimal(filter, locale);
-            return filterBigDecimal != null && getPredicate().test(toBigDecimal(value, locale), filterBigDecimal);
+        if (!(value instanceof Comparable)) {
+            throw new IllegalArgumentException("Value should be a java.lang.Comparable");
         }
-        else {
-            return getPredicate()
-                    .test(StringFilterConstraint.toString(value, locale),
-                            StringFilterConstraint.toString(filter, locale));
+        if (!filter.getClass().isAssignableFrom(value.getClass())) {
+            throw new IllegalArgumentException("Filter class should be assignable from value class. Forgot to add a converter?");
         }
-    }
-
-    static BigDecimal toBigDecimal(Object object, Locale locale) {
-        if (object instanceof BigDecimal) {
-            return (BigDecimal) object;
-        }
-        if (object instanceof Number) {
-            return new BigDecimal(object.toString());
-        }
-        if (object instanceof String) {
-            try {
-                DecimalFormat decimalFormat = (DecimalFormat) NumberFormat.getInstance(locale);
-                decimalFormat.setParseBigDecimal(true);
-                return (BigDecimal) decimalFormat.parseObject((String) object);
-            }
-            catch (ParseException e) {
-                return null;
-            }
+        if (value instanceof String) {
```

Comment:
I was wondering if we cannot just drop the special string handling.

Reply:
I'm trying to figure out something about that whole thing... Somehow I feel we should always fallback on String comparison if value.class != filter.class. But yeah, I think we should drop it, specially that it makes exact filter irrelevant since lowercase is used

Reply:
We would have the fallback on line 48, only without to lowercase conversion (which I think is fine to leave out in this case).

Reply:
Yes

Reply:
Yeah, see also #8355